framework/src/modules/pos/endpoint.ts

```diff
@@ -222,15 +222,28 @@ export class PoSEndpoint extends BaseEndpoint {
 		ctx: ModuleEndpointContext,
 	): Promise<{ validators: ValidatorAccountEndpoint[] }> {
 		validator.validate<GetValidatorsByStakeRequest>(getValidatorsByStakeRequestSchema, ctx.params);
+		const limit = ctx.params.limit as number;
```

Comment:
with the above validation, i think we don't need `as number` conversion here?
(also below, you treat limit as `number|undefined`)

Reply:
Without casting the type deferred is `{}` and the linter complains
`Invalid type "{}" of template literal expression.`
`Argument of type '{}' is not assignable to parameter of type 'number'`

The casting below is redundant though, I'll clean it up.

Reply:
true that is strange 🤔  it seems like if it's interface, it doesn't work for some reason... we can debug this in other issue.
the casted type should be `number | undefined` per schema tho

Reply:
Created issue #7953